rotkehlchen/history/events/utils.py

```diff
@@ -55,7 +55,7 @@ def create_event_identifier(
     location makes a unique event identifier. If this is not available, the location,
     timestamp, asset, and balance must all be combined to ensure a unique identifier.
     """
-    if unique_id is not None:
+    if unique_id is not None and len(unique_id) > 0:
```

Comment:
Why this check? An empty string is a valid identifier. If you give it then it's your problem, should not give a an empty string as an identifier. Same as if you give the identifier 'test' etc.

Same question below